libnavigation-ui/src/main/java/com/mapbox/navigation/ui/voice/SpeechPlayerProvider.java

```diff
@@ -114,9 +114,15 @@ void onDestroy() {
   }
 
   void onSpeechPlayerStateChanged(@NonNull SpeechPlayerState speechPlayerState) {
-    this.speechPlayerState = speechPlayerState;
+    if (this.speechPlayerState == SpeechPlayerState.OFFLINE_PLAYER_INITIALIZING
+        && speechPlayerState == SpeechPlayerState.OFFLINE_PLAYER_INITIALIZED) {
```

Comment:
When receiving `OFFLINE_PLAYER_INITIALIZED` as new state, it only updates the state to `IDLE` when the old state is still `OFFLINE_PLAYER_INITIALIZING`. Because the TTS engine initialization is only slightly delayed. If the old state is already changed, it means the online player has played voice guidance.